lib/FIR/statements.cc

```diff
@@ -126,19 +125,25 @@ LoadInsn::LoadInsn(Statement *addr) : address_{addr} {
   CHECK(GetAddressable(addr));
 }
 
-StoreInsn::StoreInsn(Statement *addr, Statement *val)
-  : address_{GetAddressable(addr)} {
+// Store ctors
+StoreInsn::StoreInsn(QualifiedStmt<Addressable_impl> addr, BasicBlock *val)
+  : address_{addr}, value_{val} {
   CHECK(address_);
-  if (auto *value{GetAddressable(val)}) {
-    value_ = value;
-  } else {
-    auto *expr{GetApplyExpr(val)};
-    CHECK(expr);
-    value_ = expr;
-  }
+  CHECK(val);
+}
+StoreInsn::StoreInsn(QualifiedStmt<Addressable_impl> addr, Value val)
+  : address_{addr}, value_{val} {
+  CHECK(address_);
```

Comment:
is a null `val` legal here?

Reply:
At this point, we can store a `Nothing`.  That may be useful will this gets built out, but should eventually be phased out.